use-shopping-cart/src/@types/use-shopping-cart/index.d.ts

```diff
@@ -0,0 +1,105 @@
+declare module 'use-shopping-cart' {
+
+    export interface ProviderProps {
+        children: JSX.Element,
+        stripe: any,
+        successUrl: string,
+        cancelUrl: string,
+        currency: string,
+        language: string,
+        billingAddressCollection: boolean,
+        allowedCountries: null | string[]
+    }
+
+    /**
+	 * Context provider to interact with Stripe API
+	 * @param children The Entry point to your app
+	 * @param stripe The stripe object
+	 * @param successUrl The redirect url for a successful sale
+	 * @param cancelUrl The redirect url for a cancelled sale
+	 * @param currency The preferred currency
+	 * @param language The language to be used in the checkout
+	 * @param billingAddressCollection Should the billing address be collected at the checkout. Defaults to false
+     * @param allowedCountries The allowed countries
+	 */
+    export const CartProvider: (props: ProviderProps) => JSX.Element;
+
+    export interface Product {
+        /** 
+         * The name of the product
+         */
+        name?: string;
+        /**
+         * The product sku
+         */
+        sku: string;
+        /**
+         * The price of the product
+         */
+        price: number;
+        /**
+         * An image of the product
+         */
+        image?: string;
+        /**
+         * The currency of the product
+         */
+        currency?: string;
+    }
+
+    export type CartDetails = {
+        [key: string]: any
+    };
+
+    export interface ShoppingCartUtilities {
+        /**
+         * Add an item to the cart
+         * @param product The product to add to the cart
+         */
+        addItem: (product: Product) => void, 
+        /**
+         * Remove a cart item
+         * @param sku The item to remove sku
+         */
+        removeCartItem: (sku: string) => void,
+        /**
+         * Reduce the quantity of items by one in the cart
+         * @param sku The sku of the item to reduce quantity by one
+         */
+        reduceItemByOne: (sku: string) => void,
+        /**
+         * Calculates the total price of the cart items
+         */
+        totalPrice: () => string;
+        /**
+         * The number of items in the cart
+         */
+        cartCount: (items: number) => void;
+        /**
+         * Cart details is an object with skus of the items in the cart as keys and details of the items as the value,
+         */
+        cartDetails: (cartEntries: CartDetails) => void;
```

Comment:
`cartDetails` is just an object not a function

Reply:
Cool